### ISSUE TITLE
E2E Testpocalypse

### ISSUE THREAD

**Issue by jaril** [RecordReplay/devtools#6046]
## Todo

- [x] Fix `breakpoints-02.js` (https://github.com/RecordReplay/devtools/pull/6047)
- [x] Fix `breakpoints-05.js` (https://github.com/RecordReplay/devtools/pull/6047)
- [x] Fix `settings.js`

### `console_eval` [(link)](https://app.replay.io/recording/7fe6169b-406d-4623-a615-cb276f1ba819?point=53221042802395999061821018744031678&time=28012&hasFrames=false)
- Left comments in the replay. Basically the test evaluates something to show a function, but the function shows up as `Function {...}` instead of `function updateNumber`.
- Maybe it's related to: https://github.com/RecordReplay/devtools/issues/5570 ?

### `stepping-03` [(link)](https://app.replay.io/recording/11f6ee0a-a190-4a80-80c0-eb02767baea9?point=18497557560050717908253089699202084&time=7532&hasFrames=true)
- This thing's complaining about a similar codemirror problem to breakpoints-02. I bet it hasn't rebased yet. Let's try running it again and maybe there's no problem after all.

### `stepping-05` [(link)](https://app.replay.io/recording/5590d492-0e6d-40d1-ae3a-cb6b32d96485)
- Already have [an issue](https://github.com/RecordReplay/devtools/issues/5862) filed for this one.
- This adds an event breakpoint to click. The click event shows up in the console, but without its object value (e.g. `click { ... }`).

### `logpoint-03` [(link)](https://app.replay.io/recording/ac3ab545-ca68-494c-abec-6dfaef92ae2f)
- This test tries to add a click event breakpoint and waits for it to show up in the console.
- It never shows up, so it times out and fails.

### `breakpoints-02` [(link)](https://app.replay.io/recording/c2192037-e606-46ac-a5ec-f82a6dd1cf34)
- This looks like it's running into the same error that was supposed to already have been solved [here](https://github.com/RecordReplay/devtools/pull/6013).
- It looks like there's an editor in the BreakpointOptions component that's also waiting for the source editor (see comment). 
- Add a guard here similar to `waitForEditor` and we should be ready to roll.

### `breakpoints-05` [(link)](https://app.replay.io/recording/7c8add4a-4676-4da3-93a9-2f0ccda5b68c)
- Looks like it's running into the same class of error as breakpoint-02. There's a race condition where the breakpoint gets loaded first before CodeMirror has had a chance to load the source editor.

### `logpoints-02` [(link)](https://app.replay.io/recording/a74d6fd0-2981-4db8-b399-4b6bfed49f6f)
- This times out because there was an unexpected socket error.
- Probably a valid intermittent.

### `inspector-05` [(link)](https://app.replay.io/recording/9602a874-a649-4f51-9f79-7ca4f9e49c79)
- Already have [an issue](https://github.com/RecordReplay/devtools/issues/6037) filed for this one.
- Holger says probably too difficult to debug until after sourcemaps makes its way in.

## `inspector-07` [(link)](https://app.replay.io/recording/239110f9-0b8a-4303-8220-23ac1eccc26c)
- Socket error. Definitely intermittent.

### `object_preview-01` [(link)](https://app.replay.io/recording/603ea1b1-6125-4033-af15-3c2a5fcc77f1)
- This thing doesn't open for me welp.

### `object_preview-02` [(link)](https://app.replay.io/recording/cea44809-3ac6-437e-afc4-ec71d4cfba45?point=39266744995177107933151750088820204&time=22210&hasFrames=false)
- This test tries to expand the this object to see functions in the scope.
- It hits expand, then waits for a particular variable.
- That variable is never shown because it's stuck in `Loading...` until it times out.

### `object_preview-03` [(link)](https://app.replay.io/recording/25440568-1ddc-403e-9ec2-b506ed43bd1d)
- This thing doesn't open up for me welp.

### `settings` [(link)](https://app.replay.io/recording/076044d0-6463-4914-b7e4-13d984607810)
- Looks like it's not getting an `unsubscribedEmailTypes` object from the hook
- We're getting a null-ish value for the viewer (see comment).

### `highlighter` [(link)](https://app.replay.io/recording/66255c82-d1ee-46e8-b464-06c73f5bdabe)
- Test never ran because stuck in loading screen.

### `invalid-recording-01.ts`  (no link)

**Comment by jaril** [comment]
@hbenl might need your help to figure out why `settings.js` is acting up :/

**Comment by jaril** [comment]
More info on `settings.js`: https://ui.honeycomb.io/replay/datasets/backend/result/iNDif63RHxn/a/3fveKN7U4W1?hideCompare&useStackedGraphs

**Comment by jaril** [comment]
Figured out the problem. Looks like it was caused by https://github.com/RecordReplay/devtools/pull/5968.

@jcmorrow and I were wondering how we were testing auth-related things with our e2e tests in the first place. Turns out, we're not — `settings.js` really just tested the Experimental page.

We didn't notice things failing before because they were failing silently. When #5968 was merged, we started doing things with auth-related parts of the code that fails loudly. Specifically, the Preferences tab accesses `unsubscribedEmailTypes` assuming that it's an object, when [it's actually falsy](https://github.com/RecordReplay/devtools/blob/master/src/ui/components/shared/UserSettingsModal/PreferencesSettings.tsx#L26).

We could force the old flow that the test was testing. But since it's not really testing much at this point, I'm thinking of just deleting it. 

Thoughts? @hbenl
